sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareFileItem.cs

```diff
@@ -3,36 +3,53 @@
 
 #pragma warning disable SA1402  // File may only contain a single type
 
+using System.ComponentModel;
+
 namespace Azure.Storage.Files.Shares.Models
 {
     /// <summary>
     /// Describes a file or directory returned by
-    /// <see cref="ShareDirectoryClient.GetFilesAndDirectoriesAsync"/>.
+    /// <see cref="ShareDirectoryClient.GetFilesAndDirectoriesAsync(ShareDirectoryGetFilesAndDirectoriesOptions, System.Threading.CancellationToken)"/>.
     /// </summary>
     public class ShareFileItem
     {
+        internal ShareFileItem() { }
+
         /// <summary>
         /// Gets a value indicating whether this item is a directory.
         /// </summary>
-        public bool IsDirectory { get; }
+        public bool IsDirectory { get; internal set; }
 
         /// <summary>
         /// Gets the name of this item.
         /// </summary>
-        public string Name { get; }
+        public string Name { get; internal set; }
+
+        /// <summary>
+        /// Gets the ID.
+        /// </summary>
+        public string Id { get; internal set; }
+
+        /// <summary>
+        /// Gets the properties of this item.
+        /// </summary>
+        public ShareFileItemProperties Properties { get; internal set; }
+
+        /// <summary>
+        /// Gets the file attributes.
+        /// </summary>
+        public NtfsFileAttributes? FileAttributes { get; internal set; }
+
+        /// <summary>
+        /// Gets the permission key.
+        /// </summary>
+        public string PermissionKey { get; internal set; }
 
         /// <summary>
         /// Gets an optional value indicating the file size, if this item is
         /// a file.
         /// </summary>
-        public long? FileSize { get; }
-
-        internal ShareFileItem(bool isDirectory, string name, long? fileSize = null)
```

Comment:
in my opinion it's better to leave ctor rather than make properties mutable.
and best if all ctor parameters are not optional.

the reason is that I started to notice plenty of places that do some mappings in our code in general.
problem is that if new property is introduced then it's a challenge to find all of them and update accordingly.
having ctor to take props in has advantage that compiler will highlight all the places that require update instantly.
therefore I'd call to abandon the pattern of props that are mutable internally in favor of ctors.

Reply:
didn't have to wait long to find another data point supporting this https://github.com/Azure/azure-sdk-for-net/pull/21550#discussion_r644960831

Reply:
fixed.

Reply:
please make all props readonly. let's not leave backdoor.

Reply:
done.